rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,400 @@
+                f'to {to_asset.identifier}. But from_asset is not supported in alchemy',
+            )
+            return ZERO_PRICE
+
+        if 'symbol' in asset_identifier:
+            result = self._query(
+                payload={'symbols': asset_identifier['symbol']},
+                endpoint='by-symbol',
+            )
+        else:
+            result = self._query(
+                payload={'addresses': [asset_identifier]},
+                endpoint='by-address',
+            )
+        usd_price = self._deserialize_current_price(
+            result=result['data'],
+            asset_identifier=asset_identifier,
+            from_asset=from_asset,
+            to_asset=to_asset,
+        )
+        if usd_price == ZERO_PRICE or to_asset == A_USD:
+            return usd_price
+
+        # The price is in USD, but we need to convert it to the target currency.
+        rate_price = Inquirer.find_price(from_asset=A_USD, to_asset=to_asset)
+        return Price(usd_price * rate_price)
+
+    def query_historical_price(
+            self,
+            from_asset: Asset,
+            to_asset: Asset,
+            timestamp: Timestamp,
+    ) -> Price:
+        """
+        Docs: https://docs.alchemy.com/reference/get-historical-token-prices
+
+        May raise:
+        - RemoteError if there is a problem querying Alchemy
+        - NoPriceForGivenTimestamp if no price could be found
+        - PriceQueryUnsupportedAsset if either from_asset or to_asset are not supported
+        """
+        try:
+            from_asset = from_asset.resolve_to_asset_with_oracles()
+            to_asset = to_asset.resolve_to_asset_with_oracles()
+        except UnknownAsset as e:
+            raise PriceQueryUnsupportedAsset(e.identifier) from e
+
+        if (price_cache_entry := GlobalDBHandler.get_historical_price(
+            from_asset=from_asset,
+            to_asset=to_asset,
+            timestamp=timestamp,
+            max_seconds_distance=HOUR_IN_SECONDS * 6,
+            source=HistoricalPriceOracle.ALCHEMY,
+        )):
+            return price_cache_entry.price
+
+        try:
+            asset_identifier = self._get_asset_identifier(from_asset)
+        except UnsupportedAsset as e:
+            log.warning(
+                f'Tried to query alchemy historical price from {from_asset.identifier} '
+                f'to {to_asset.identifier}. But from_asset is not supported in alchemy',
+            )
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+            ) from e
+
+        payload: dict[str, str | int] = {
+            'startTime': timestamp - HOUR_IN_SECONDS,
+            'endTime': timestamp + HOUR_IN_SECONDS,
+            'interval': '1h',
+        }
+        result = self._query(
+            endpoint='historical',
+            payload={**payload, **asset_identifier},
```

Comment:
I guess if you go with the named tuple approach you would need to implement the ** operator, or do a to_dict() function.